### PR TITLE
Streaming API (Capturing)

### PR DESCRIPTION
**Pull request by adriangranados**
## Type of change 

Implements the Streaming API (for capturing) that allows clients to:

- Get the list of supported frequencies for capturing
- Configure the desired channels (frequencies) and their widths, and the dwell time, per interface, that will be used for capturing
- Start a capture
- Stop a capture

As part of these changes, wlanpi-core will now stage the WLAN interfaces at startup. Each monitor interface will be named wlanpiN where _N_ is the index of the adapter's phy. E.g., phy1 -> wlanpi1

* [ ] Bug fix (non-breaking change that fixes something)
* [ ] Documentation update (readme, changelog, man page, etc.)
* [X] New feature (non-breaking change adding functionality)
* [ ] Breaking change (fix or feature changing existing functionality)
* [ ] Code quality improvement (refactor, performance improvements)
* [ ] Other (please specify):

## Testing

The new Streaming API was tested using a Python client that exercises the API and the WLAN Pi Go. More testing is recommended, especially on devices with multiple interfaces (which should be supported).

- [ ] Did you add new automated tests? If yes, explain which edge cases are covered.
- [ ] Does this change affect any existing tests? If so, how did you adjust them?
- [ ] Please provide test run results or screenshots if applicable.

## Checklist

 

* [X] I have read the [contribution guidelines and policies](https://github.com/WLAN-Pi/.github/blob/main/docs/contributing.md)
* [X] I have targeted this PR against the correct git branch (this can vary depending on the default branch of the repo)
* [ ] I ran the test suite and verified it succeeded
* [ ] I linked a GitHub issue to this PR (in the next section). 
* [ ] I have updated the changelog (if applicable)
* [ ] I have added or updated the documentation (if applicable)

## Related Issues/PRs

- This PR is related to issue #91

### CODE REVIEW

File: wlanpi_core/streaming/connection_manager.py [hunk 1]
```diff
@@ -1,52 +1,350 @@
...
+                "error",
+                "FREQ_FETCH_FAILED",
+                f"Failed to fetch supported frequencies: {e}",
+            )
+
+    async def start_streaming(
+        self,
+        websocket: WebSocket,
+        interfaces: list[str],
+        pcap_filter: str | None = None,
```
Comment (joshschmelzle) [inline wlanpi_core/streaming/connection_manager.py]: Newer union type syntax (str | None) is not supported in Py3.9 which we still have to support.

Comment (joshschmelzle) [reply wlanpi_core/streaming/connection_manager.py]: Fixed in https://github.com/WLAN-Pi/wlanpi-core/pull/97/commits/b90f341af9036f9e3938074a50842ffd81ce568a.
